internal/service/advancedcluster/resource_upgrade.go

```diff
@@ -40,9 +46,15 @@ func getUpgradeFlexToDedicatedRequest(state, patch *admin.ClusterDescription2024
 	if oldProviderName != flexcluster.FlexClusterType || newProviderName == flexcluster.FlexClusterType {
 		return nil
 	}
-	return &admin.AtlasTenantClusterUpgradeRequest20240805{
+	req := admin.AtlasTenantClusterUpgradeRequest20240805{
 		Name:             state.GetName(),
 		ClusterType:      state.ClusterType,
 		ReplicationSpecs: patch.ReplicationSpecs,
 	}
+
+	// checking for state value as a flex cluster can already have backup enabled
+	if state.GetBackupEnabled() || patch.GetBackupEnabled() {
+		req.BackupEnabled = conversion.Pointer(true)
+	}
```

Comment:
```suggestion
		req.BackupEnabled = conversion.Pointer(state.GetBackupEnabled() || patch.GetBackupEnabled())
```

nit: i suppose this is equivalent (only difference that we send explicit false if not enabled)

Reply:
correct, prefer to just keep upgrade request more concise for the case backup is not enabled (more common)